lexicon/conversion.py

```diff
@@ -94,50 +128,53 @@ def tasks(self):
-            old_lexicon = ET.fromstring(lexicon_file.read())
+        lex = lexicon.Lexicon()
+        lex.load(self.bliss_lexicon.get_path())
 
-        root = ET.Element("lexicon")
-        root.append(old_lexicon.find("phoneme-inventory"))
+        orth2lemmata = collections.defaultdict(list)
 
-        lemmas = collections.OrderedDict()
-        for lemma in old_lexicon.findall("lemma"):
-            if "special" in lemma.attrib:
-                root.append(lemma)
-            elif len(lemma.findall("orth")) != 1:
-                root.append(lemma)
-            else:
-                orth = lemma.find("orth").text
+        for lemma in lex.lemmata:
 
-                if orth not in lemmas:
-                    lemmas[orth] = {"phon": set(), "synt": set(), "eval": set()}
+            if lemma.special:
+                continue
 
-                lemmas[orth]["phon"].update(e.text for e in lemma.findall("phon"))
-                lemmas[orth]["synt"].update(
-                    lemma.findall("synt")
-                )  # as synt can contain sub elements
-                lemmas[orth]["eval"].update(e.text for e in lemma.findall("eval"))
+            num_orths = len(lemma.orth)
+            if num_orths < 1:
+                continue
 
-        for orth, lemma in lemmas.items():
-            el = ET.SubElement(root, "lemma")
+            if num_orths > 1 and not self.merge_multi_orths_lemmata:
+                continue
 
-            o = ET.SubElement(el, "orth")
-            o.text = orth
+            orth2lemmata[lemma.orth[0]].append(lemma)
 
-            for phon in lemma["phon"]:
-                p = ET.SubElement(el, "phon")
-                p.text = phon
+        for orth, lemmata in orth2lemmata.items():
 
-            for synt in lemma["synt"]:
-                el.append(synt)
+            if len(lemmata) < 2:
+                continue
 
-            for eval in lemma["eval"]:
-                ev = ET.SubElement(el, "eval")
-                ev.text = eval
+            final_lemma = lemmata[0]
 
-        with uopen(self.out_bliss_lexicon.get_path(), "w") as lexicon_file:
-            lexicon_file.write('<?xml version="1.0" encoding="utf-8"?>\n')
-            lexicon_file.write(ET.tostring(root, "unicode"))
+            for lemma in lemmata[1:]:
+
+                for orth in lemma.orth:
+                    if orth not in final_lemma.orth:
+                        final_lemma.orth.append(orth)
+
+                for phon in lemma.phon:
+                    if phon not in final_lemma.phon:
+                        final_lemma.phon.append(phon)
+
+                if final_lemma.synt is None and lemma.synt is not None:
+                    final_lemma.synt = lemma.synt
+
+                for eval in lemma.eval:
+                    if eval not in final_lemma.eval:
+                        final_lemma.eval.append(eval)
+
+                if lemma in lex.lemmata:
```

Comment:
why is this check necessary? Wouldn't the lemma always be in lex.lemmata? You just picked it out from there.